pkgs/development/tools/pnpm/fetch-deps/default.nix

```diff
@@ -22,6 +22,15 @@ in
       pnpmWorkspaces ? [ ],
       prePnpmInstall ? "",
       pnpmInstallFlags ? [ ],
+
+      # This is the version of the fetchDeps script.
+      # New bug fixes that would change the final hash must only affect a newer version,
+      # hash of the same version must not change.
+      # Users are expected to opt-in to newer versions by setting this argument.
+      # Version history:
+      # - 1: Initial version, nothing special
+      # - 2: Ensure consistent permissions, fixes https://github.com/NixOS/nixpkgs/issues/422889
+      version ? 1,
```

Comment:
IMO, this should be set to some invalid value (i.e. `null`) and we should have an assertion that this must be set by the caller.

We would obviously update every occurrence of fetchDeps in nixpkgs accordingly here.

We can then use this same mechanism to deprecate and remove versions at some point.

Reply:
> IMO, this should be set to some invalid value (i.e. null) and we should have an assertion that this must be set by the caller.

While this may result of in cleaner code overall and make sure that everyone knows about `fetcherVersion`, I think this would be too annoying for package maintainers

> We would obviously update every occurrence of fetchDeps in nixpkgs accordingly here.

Yes, but we need to find PRs with new pnpm packages and packages in other repos would break

> We can then use this same mechanism to deprecate and remove versions at some point.

Hmm this is a good point where I'm also leaning towards defaulting to an invalid `null` version, I'll think about it